File: accsaber-api/src/main/java/de/ixsen/accsaber/api/mapping/CategoryMapper.java
package de.ixsen.accsaber.api.mapping;

import de.ixsen.accsaber.api.dtos.CategoryDto;
import de.ixsen.accsaber.database.model.Category;
import org.mapstruct.InheritInverseConfiguration;
import org.mapstruct.Mapper;
import org.mapstruct.Mapping;

import java.util.List;

@Mapper(componentModel = "spring")
public interface CategoryMapper {

    @Mapping(target = "name", source = "categoryName")
    CategoryDto map(Category category);

    @InheritInverseConfiguration
    Category map(CategoryDto categoryDto);

    List<CategoryDto> map(List<Category> categories);
}

File: accsaber-api/src/main/kotlin/de/ixsen/accsaber/api/mapping/CategoryMapper.kt
package de.ixsen.accsaber.api.mapping

import de.ixsen.accsaber.api.dtos.CategoryDto
import de.ixsen.accsaber.database.model.Category
import org.mapstruct.InheritInverseConfiguration
import org.mapstruct.Mapper
import org.mapstruct.Mapping

@Mapper(componentModel = "spring")
interface CategoryMapper {
    @Mapping(target = "categoryName", source = "categoryName")
    fun map(category: Category): CategoryDto

    @InheritInverseConfiguration
    fun map(categoryDto: CategoryDto): Category
    fun map(categories: List<Category>): List<CategoryDto>
}